### PR TITLE
Arm64/Sve: Handle ConvertTo* APIs properly

### PR DESCRIPTION
**Pull request by kunalspathak**
We were hitting an assert in these tests because we were not handling the wrapping of these intrinsics inside `ConditionalSelect.

- Make sure that `ConvertTo*` APIs get contained inside `ConditionalSelect`. To do that we now store the auxiliary type as baseType of return vector and if that matches with the wrapped mask operation, then we contain the intrinsics.
- Fix the codegen depending on the variant of the operation (see below)

`ConvertToInt32` implements https://docsmirror.github.io/A64/2023-06/fcvtzu_z_p_z.html for following variants:
  - Single-precision to 32-bit  - `FCVTZS <Zd>.S, <Pg>/M, <Zn>.S`
  - Double-precision to 32-bit  - `FCVTZS <Zd>.S, <Pg>/M, <Zn>.D`

`ConvertToUInt32` implements https://docsmirror.github.io/A64/2023-06/fcvtzs_z_p_z.html for following variants:
- Single-precision to 32-bit  - `FCVTZU <Zd>.S, <Pg>/M, <Zn>.S`
- Double-precision to 32-bit  - `FCVTZU <Zd>.S, <Pg>/M, <Zn>.D`

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/arm64-contrib  @amanasifkhalid  PTAL

@ebepho - For `ConvertTo*Int64`, you need to follow along this path.

File: src/coreclr/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -3347,14 +3347,35 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
...
+                        }
+                        else
+                        {
+                            // Else check if this operation has an auxilary type that matches the
+                            // mask size.
+                            GenTreeHWIntrinsic* embOp = op2->AsHWIntrinsic();
+
+                            // For now, make sure that we get here only for intrinsics that we are
+                            // sure about to rely on auxilary type's size.
+                            assert((embOp->GetHWIntrinsicId() == NI_Sve_ConvertToInt32) ||
```
Comment (amanasifkhalid) [inline src/coreclr/jit/lowerarmarch.cpp:3369]: Do you intend for this assert to go away once @ebepho implements `ConvertToInt64`, or should we keep checking for intrinsics with different base/return types? If the latter, I wonder if we ought to add a flag for these types of intrinsics.

File: src/coreclr/jit/lowerarmarch.cpp [hunk 2]
```diff
@@ -3347,14 +3347,35 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
...
+                        if (maskSize == operSize)
+                        {
+                            // If the size of baseType of operation matches that of maskType, then contain
+                            // the operation
+                            MakeSrcContained(node, op2);
+                            op2->MakeEmbMaskOp();
+                        }
+                        else
+                        {
+                            // Else check if this operation has an auxilary type that matches the
```
Comment (amanasifkhalid) [inline src/coreclr/jit/lowerarmarch.cpp:3363]:
```suggestion
                            // Else check if this operation has an auxiliary type that matches the
```

File: src/coreclr/jit/lowerarmarch.cpp [hunk 3]
```diff
@@ -3347,14 +3347,35 @@ void Lowering::ContainCheckHWIntrinsic(GenTreeHWIntrinsic* node)
...
+                            // mask size.
+                            GenTreeHWIntrinsic* embOp = op2->AsHWIntrinsic();
+
+                            // For now, make sure that we get here only for intrinsics that we are
+                            // sure about to rely on auxilary type's size.
```
Comment (amanasifkhalid) [inline src/coreclr/jit/lowerarmarch.cpp:3368]:
```suggestion
                            // sure about to rely on auxiliary type's size.
```

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks for fixing this!

File: src/coreclr/jit/lowerarmarch.cpp (hunk 1 above)
Comment (kunalspathak) [reply src/coreclr/jit/lowerarmarch.cpp:3369]: For now, we will update the assert, but may be later, we might consider adding a flag. For now, I want to see if there are more intrinsic categories that falls under this.

**Pull request merged**